### DISCUSSION TITLE
Stochastic modeling - time dependent probabilities/sampling

### DISCUSSION THREAD

**Discussion by DarioSlaifsteinSk** [infiniteopt/InfiniteOpt.jl#343] [Q&A]
hi!
Researching about stochastic modeling, I stumbled across the RFO paper [Pulsipher (2022)](https://linkinghub.elsevier.com/retrieve/pii/S0098135422001922) with [Case Studies](https://github.com/zavalab/JuliaBox/tree/master/RandomFieldOptCases). 
And I couldn't help to notice that the syntax is not the same as in the [Pandemic Control tutorial](https://infiniteopt.github.io/InfiniteOpt.jl/stable/examples/Optimal%20Control/pandemic_control/).

In the first one, uncertainty is not modeled with `@infinite_parameter`, but all the scenarios are generated, and then all the vars and constraints are repeated for each scenario. Maybe the package wasn't mature back then, but how would the formulation differ?

Because generating the inputs saves some complications (domain dependencies and such), it also doesn't let you use some new features like changing the pdf to estimate the expectation. This would be relevant in cases where the Random Field is time-dependent or a subset of the primary domain (random events).

**Comment by pulsipher** [comment]
Hi @DarioSlaifsteinSk, good work finding the RFO paper!

InfiniteOpt currently only support static uncertainty (e.g., time agnostic). To handle time-dependent uncertainty, we would need to support an infinite parameter this is a function of another (e.g., `xi(t)`). 

This is on the TODO list (#95), but I haven't had time to pursue it yet. Part of the problem is that coming up with effective, generic solution strategies for these problems is hard and basic one used in the RFO paper has its limitations. PRs/ideas welcome :)

In the meantime, the best approach is the use the coding approach from the RFO paper or you can check out SDDP.jl: https://sddp.dev/stable/

**Reply by DarioSlaifsteinSk** [reply]
ha! seems that I've hit the nail on this one!
Cool I'll try some implementations and get back with the results.

**Comment by DarioSlaifsteinSk** [comment]
Well 6 months later.
This is what I came up with:

https://github.com/DarioSlaifsteinSk/StochMobilityMCES-ITEC24

[D. Slaifstein, A. M. Agudin, G. R. C. Mouli, L. Ramírez-Elizondo and P. Bauer, "Stochastic Mobility Integration into Residential Energy Hubs," 2024 IEEE International Conference on Electrical Systems for Aircraft, Railway, Ship Propulsion and Road Vehicles & International Transportation Electrification Conference (ESARS-ITEC), Naples, Italy, 2024, pp. 1-7, doi: 10.1109/ESARS-ITEC60450.2024.10819794.](https://ieeexplore-ieee-org.tudelft.idm.oclc.org/document/10819794)

**Reply by pulsipher** [reply]
Thanks for sharing!
